### PR TITLE
Add time to experiment termination conditions

### PR DESCRIPTION
**Pull request by mleot**
# Description

This change is to add the ability to stop an experiment by setting a time delta as a termination condition. There is no current issue for this that I am aware of, but I can make one if needed. I would like to be able to do this to simulate an experiment only up until a given time. 

Fixes # (issue)

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [x] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [x] All tests pass: `$ python run-tests.py --all` (or `$ nox -s tests`)
- [x] The documentation builds: `$ python run-tests.py --doctest` (or `$ nox -s doctests`)

You can run integration tests, unit tests, and doctests together at once, using `$ python run-tests.py --quick` (or `$ nox -s quick`).

## Further checks:

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works

### CODE REVIEW

**Review by valentinsulzer — COMMENTED** [review]
Looks good to me. @brosaplanella can you confirm this isn't duplicating the functionality of the `start_time` that you added?

File: pybamm/experiment/experiment.py [hunk 1]
```diff
@@ -169,6 +169,15 @@ def read_termination(termination):
...
                 termination_dict["voltage"] = (float(end_discharge_V), "V")
+            elif term.endswith("s"):
+                end_time_s = term.split("s")[0]
+                termination_dict["time"] = (float(end_time_s), "s")
+            elif term.endswith("min"):
+                end_time_s = term.split("min")[0]
+                termination_dict["time"] = (float(end_time_s) * 60, "s")
+            elif term.endswith("h"):
+                end_time_s = term.split("h")[0]
+                termination_dict["time"] = (float(end_time_s) * 3600, "s")
```
[the remark this answers is not in the document]
Comment (kratman) [reply pybamm/experiment/experiment.py:180]: I created #4113 to track this, @mleot let me know if you want to take that issue as well

**Comment by kratman** [comment]
> For Lychee the whole projects page of LLNL is down, but the main page is still OK. Probably just a random outage

Failure is gone now

File: pybamm/experiment/experiment.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mleot) [reply pybamm/experiment/experiment.py:180]: Thanks for creating that issue. It sounds like there should be another change applied to use the function that Valentin mentioned in that issue. I can take that issue if nobody else wants to take it. Maybe next week I can take a stab at applying the fix.

**Pull request merged**